crates/bcr-wdc-balance-collector/src/main.rs

```diff
@@ -0,0 +1,83 @@
+use chrono::DurationRound;
+use std::str::FromStr;
+use tokio::signal;
+use tracing_subscriber::{filter::LevelFilter, prelude::*};
+
+#[derive(Debug, serde::Deserialize)]
+struct MainConfig {
+    log_level: String,
+    appcfg: bcr_wdc_balance_collector::AppConfig,
+    interval_minutes: u64,
+}
+
+#[tokio::main]
+async fn main() {
+    let settings = config::Config::builder()
+        .add_source(config::File::with_name("config.toml"))
+        .add_source(config::Environment::with_prefix("BALANCE_COLLECTOR"))
+        .build()
+        .expect("Failed to build wildcat config");
+
+    let maincfg: MainConfig = settings
+        .try_deserialize()
+        .expect("Failed to parse wildcat config");
+
+    tracing_log::LogTracer::init().expect("LogTracer init");
+    let level_filter = LevelFilter::from_str(&maincfg.log_level).expect("log level");
+    let stdout_log = tracing_subscriber::fmt::layer().with_filter(level_filter);
+    let subscriber = tracing_subscriber::registry().with(stdout_log);
+    tracing::subscriber::set_global_default(subscriber)
+        .expect("tracing::subscriber::set_global_default");
+
+    let controller = bcr_wdc_balance_collector::AppController::new(maincfg.appcfg)
+        .await
+        .expect("AppController init");
+
+    let rounder = chrono::Duration::minutes(maincfg.interval_minutes as i64);
+    loop {
+        let next_interval = next_interval(rounder);
+        tokio::select! {
+        _ = shutdown_signal() => {
+            tracing::info!("Shutdown signal received, exiting...");
+        },
+        _ = tokio::time::sleep(next_interval) => {
+            tracing::info!("Starting balance collection...");
+                controller.collect_balances(chrono::Utc::now()).await.expect("Balance collection failed");
+            }
+        }
+    }
+}
+
+async fn shutdown_signal() {
+    let ctrl_c = async {
+        signal::ctrl_c()
+            .await
+            .expect("failed to install Ctrl+C handler");
+    };
+
+    #[cfg(unix)]
+    let terminate = async {
+        signal::unix::signal(signal::unix::SignalKind::terminate())
+            .expect("failed to install signal handler")
+            .recv()
+            .await;
+    };
+
+    #[cfg(not(unix))]
+    let terminate = std::future::pending::<()>();
+
+    tokio::select! {
+        _ = ctrl_c => {},
+        _ = terminate => {},
+    }
+}
+
+fn next_interval(rounder: chrono::Duration) -> tokio::time::Duration {
```

Comment:
Does this lead to the interval always being on the full hour? (trunc to last full hour, add one hour, if already past, fallback to now+60sec?)

If so, I think a comment describing this could be helpful.

And I guess if I misinterpreted it, then a comment describing what it actually does would be even more helpful. :joy: